test5.rb

```diff
@@ -0,0 +1,136 @@
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/CyclomaticComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    c = 0
+    while c < to_a.length
+      yield(to_a[c])
+      c += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    c = 0
+    while c < to_a.length
+      yield(to_a[c], c)
+      c += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each { |item| result << item if yield(item) }
+    result
+  end
+
+  def my_all?(pattern = nil)
+    c = 0
+    statement = true
+    if block_given?
+      to_a.length.times do
+        statement = false unless yield(to_a[c])
+        c += 1
+      end
+    elsif !pattern.nil?
+      to_a.length.times do
+        statement = false if to_a[i].scan(pattern)
+      end
+    end
+    statement
+  end
+
+  def my_none?(pattern = nil)
+    c = 0
+    statement = true
+    if block_given?
+      to_a.length.times do
+        statement = false if yield(to_a[c])
+        c += 1
+      end
+    elsif !pattern.nil?
+      to_a.length.times do
+        statement = false if to_a[i].scan(pattern)
+      end
+    end
+    statement
+  end
+
+  def my_any?(pattern = nil)
+    c = 0
+    statement = false
+    if block_given?
+      length.times do
+        statement = true if yield(self[c])
+        c += 1
+      end
+    elsif !pattern.nil?
+      length.times do
+        statement = true if self[i].scan(pattern)
+      end
+    end
+    statement
+  end
```

Comment:
### my_any? should behave as follow:
- [ ] 9. returns false if the block ever returns a value false or nil::range
- [ ] 10. when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] 11. when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
- [ ] 12. when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
- [ ] 13. when a Regex is passed as an argument returns true if any of the collection matches the Regex
- [ ] 14. when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] 15. when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
- [ ] 16. when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern